docs/tendermint-core/state-id.md

```diff
@@ -0,0 +1,43 @@
+# State ID
+
```

Comment:
That's cool having this doc. Could you please add a paragraph with a rationale for having StateID? I believe it will be useful for others.

Reply:
@QuantumExplorer can you write a few words of rationale please?

Reply:
Good point, will do it.